viewer/lib/controls.js

```diff
@@ -0,0 +1,909 @@
+            this.state = STATE.TOUCH_DOLLY_PAN
+            break
+          case THREE.TOUCH.DOLLY_ROTATE:
+            if (this.enableTouchZoom === false && this.enableTouchRotate === false) return
+            this.handleTouchStartDollyRotate(event)
+            this.state = STATE.TOUCH_DOLLY_ROTATE
+            break
+          default:
+            this.state = STATE.NONE
+        }
+        break
+      default:
+        this.state = STATE.NONE
+    }
+    if (this.state !== STATE.NONE) {
+      this.dispatchEvent(this.startEvent)
+    }
+  }
+
+  onTouchMove(event) {
+
+    if (this.enabled === false) return
+
+    event.preventDefault() // prevent scrolling
+    event.stopPropagation()
+
+    switch (this.state) {
+
+      case STATE.TOUCH_ROTATE:
+
+        if (this.enableTouchRotate === false) return
+
+        this.handleTouchMoveRotate(event)
+
+        this.update()
+
+        break
+
+      case STATE.TOUCH_PAN:
+
+        if (this.enableTouchPan === false) return
+
+        this.handleTouchMovePan(event)
+
+        this.update()
+
+        break
+
+      case STATE.TOUCH_DOLLY_PAN:
+
+        if (this.enableTouchZoom === false && this.enableTouchPan === false) return
+
+        this.handleTouchMoveDollyPan(event)
+
+        this.update()
+
+        break
+
+      case STATE.TOUCH_DOLLY_ROTATE:
+
+        if (this.enableTouchZoom === false && this.enableTouchRotate === false) return
+
+        this.handleTouchMoveDollyRotate(event)
+
+        this.update()
+
+        break
+
+      default:
+
+        this.state = STATE.NONE
+
+    }
+
+  }
+
+  onTouchEnd(event) {
+
+    if (this.enabled === false) return
+
+    this.handleTouchEnd(event)
+
+    this.dispatchEvent(this.endEvent)
+
+    this.state = STATE.NONE
+
+  }
+
+
+  onContextMenu(event) {
+    // Disable context menu
+    if (this.enabled) event.preventDefault()
+  }
+
+  registerHandlers() {
+    this.element.addEventListener('pointermove', this.onPointerMove.bind(this), false)
+    this.element.addEventListener('pointerup', this.onPointerUp.bind(this), false)
+    this.element.addEventListener('pointerdown', this.onPointerDown.bind(this), false)
+    this.element.addEventListener('wheel', this.onMouseWheel.bind(this), true)
+
+    this.element.addEventListener('touchstart', this.onTouchStart.bind(this), false)
+    this.element.addEventListener('touchend', this.onTouchEnd.bind(this), false)
+    this.element.addEventListener('touchmove', this.onTouchMove.bind(this), false)
+
+    this.element.ownerDocument.addEventListener('contextmenu', this.onContextMenu.bind(this), false)
+    this.element.ownerDocument.addEventListener('keydown', this.onKeyDown.bind(this), false)
+    this.element.ownerDocument.addEventListener('keyup', this.onKeyUp.bind(this), false)
+    console.log('[controls] registered handlers', this.element)
+  }
+
+  unregisterHandlers() {
+    this.element.removeEventListener('pointermove', this.onPointerMove, false)
```

Comment:
These aren't actually going to remove anything, because `this.onPointerMove.bind(this) !== this.onPointerMove`. Your best bet might be to autobind class properties in the constructor so that you can use `this.onPointerMove` everywhere and have it bound to the class.

Reply:
Should be fixed.